akka-actor/src/main/scala/akka/pattern/RetrySettings.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2025 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.pattern
+
+import java.time.Duration
+import java.util.Optional
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.FiniteDuration
+import scala.jdk.DurationConverters.JavaDurationOps
+
+sealed trait RetrySettings {}
+
+object RetrySettings {
+
+  case class RetrySettingsBuilder(attempts: Int) {
```

Comment:
More aligned with other Akka library settings: Skip the builder and instead have factory `create`/`apply` methods with overloads. If there is ambiguity, we can have factory methods with more clear names for those (like `RetrySettings.withFixedDelay(times, delay)` perhaps).

The concrete `RetrySettings` then has the `with` methods to allow changing details after the construction.


Also: no case classes here unless internal API/hidden from users, highly problematic for bin comp evolvability with the copy method and traits it automatically extends.